src/store/conversationsStore.js

```diff
@@ -428,15 +429,25 @@ const actions = {
 	},
 
 	async markConversationRead({ commit, getters }, token) {
-		const conversation = Object.assign({}, getters.conversations[token])
-		if (!conversation) {
+		if (!getters.conversations[token]) {
 			return
 		}
 
-		conversation.unreadMessages = 0
-		conversation.unreadMention = false
+		commit('updateUnreadMessages', { token, unreadMessages: 0, unreadMention: false })
+	},
 
-		commit('addConversation', conversation)
+	async markConversationUnread({ commit, dispatch, getters }, { token }) {
+		if (!getters.conversations[token]) {
+			return
+		}
+
+		try {
+			await setConversationUnread(token)
+			commit('updateUnreadMessages', { token, unreadMessages: 1 })
+			await dispatch('fetchConversation', { token })
+		} catch (error) {
```

Comment:
catching again? you are already catching in the `setConversationUnread`?
Whatever happens to be catched here is not expected and should be console.error at least.

Maybe you at least want to move the first 2 calls out of the try?

Reply:
Here I'm agree, that we could omit try-catch blocks:
`setConversationUnread` - catches errore in servise.js
`commit('updateUnreadMessages')`  - mutation, for which we have checked before that conversation exists in `if` block
`dispatch('fetchConversation')` - also catches errors by its own

Reply:
> `dispatch('fetchConversation')` - also will catch errors by its own

It re-throw it

https://github.com/nextcloud/spreed/blob/077a584c86a2c4e1942e91835647b47382738a41/src/store/conversationsStore.js#L633-L638

But it's ok to not catch exception here to me, because we don't show anything with `showError`, don't check that `setConversationUnread` was actually successful, and don't check is all action was successful on the call.